Sources/Error Handling/SKError+Extensions.swift

```diff
@@ -61,8 +65,66 @@ extension SKError {
             return ErrorUtils.storeProblemError(error: self)
 
         @unknown default:
+            switch SKError.UndocumentedCode(rawValue: self.code.rawValue) {
+            case .unhandledException:
+                if let error = self.userInfo[NSUnderlyingErrorKey] as? NSError {
+                    switch error.domain {
+                    case AMSError.domain:
+                        switch AMSError.Code(rawValue: error.code) {
+                            // See https://github.com/RevenueCat/purchases-ios/issues/1445
+                            // Cancellations sometimes show up as undocumented errors instead of regular cancellations
+                        case .paymentSheetFailed:
+                            return ErrorUtils.purchaseCancelledError(error: self)
+
+                        default: break
+                        }
+
+                    default: break
+                    }
+                }
+
+            default: break
+            }
+
             return ErrorUtils.unknownError(error: self)
         }
     }
 
 }
+
+private extension SKError {
+
+    enum UndocumentedCode: Int {
+
+        // See https://github.com/RevenueCat/purchases-ios/issues/1445
+        case unhandledException = 907
+
+    }
+
+}
+
+private enum ASDServerError {
+
+    static let domain = "ASDServerErrorDomain"
+
+    enum Code: Int {
+
+        // See https://github.com/RevenueCat/purchases-ios/issues/392
+        case currentlySubscribed = 3532
+
+    }
+
+}
+
+private enum AMSError {
+
+    static let domain = "AMSErrorDomain"
+
+    enum Code: Int {
+
+        // See https://github.com/RevenueCat/purchases-ios/issues/1445
+        case paymentSheetFailed = 6
+
+    }
+
+}
```

Comment:
I created these to better maintain these undocumented errors. Thanks for the suggestion @aboedo